subprojects/core-api/src/main/java/org/gradle/process/JavaExecSpec.java

```diff
@@ -67,6 +74,11 @@ public interface JavaExecSpec extends JavaForkOptions, BaseExecSpec {
      */
     @Optional
     @Input
+    @UpgradedProperty(
+        originalAccessors = @UpgradedAccessor(value = AccessorType.SETTER, methodName = "setMain", fluentSetter = true),
```

Comment:
🤔 What strikes me here is that "upgraded" in `@UpgradedProperty` means "this is the upgraded property, and there's an _old_ Java bean property that we are providing an upgrade for."

But in `@UpgradedAccessor` we seem to be saying that the _old_ accessor is "upgraded."

This makes it confusing what the term "upgraded" means.

I wonder if there is a better name we could use to make things more understandable to our users. Like, `@PropertyUsedToBeEager()`?

Here's a different take on this. We arrived at `@UpgradedProperty` because we are thinking about the process of the upgrade. This is natural, because currently all we have is old APIs, and lazy properties are in the future, and to get there, we need to _upgrade._ So we focus on the process of ugprading.

But when a user will look at these annotations, they will already be looking at those future lazy APIs. So the annotation should refer to the past, on what was replaced. Something like `@ReplacesDiscontinuedEagerProperty()` captures what the user would be curious about better for me.

Reply:
Renamed to `@ReplacesEagerProperty()`. Wdyt about that?

Reply:
That sounds a lot better at first blush!